uber/decorators.py

```diff
@@ -308,26 +308,31 @@ def prettify_breadcrumb(str):
 def renderable(func):
     @wraps(func)
     def with_rendering(*args, **kwargs):
-        result = func(*args, **kwargs)
-
         try:
-            result['breadcrumb_page_pretty_'] = prettify_breadcrumb(func.__name__) if func.__name__ != 'index' else 'Home'
-            result['breadcrumb_page_'] = func.__name__ if func.__name__ != 'index' else ''
-        except:
-            pass
+            result = func(*args, **kwargs)
 
-        try:
-            result['breadcrumb_section_pretty_'] = prettify_breadcrumb(get_module_name(func))
-            result['breadcrumb_section_'] = get_module_name(func)
-        except:
-            pass
+            try:
+                result['breadcrumb_page_pretty_'] = prettify_breadcrumb(func.__name__) if func.__name__ != 'index' else 'Home'
+                result['breadcrumb_page_'] = func.__name__ if func.__name__ != 'index' else ''
+            except:
+                pass
 
-        if c.UBER_SHUT_DOWN and not cherrypy.request.path_info.startswith('/schedule'):
-            return render('closed.html')
-        elif isinstance(result, dict):
-            return render(_get_template_filename(func), result)
-        else:
-            return result
+            try:
+                result['breadcrumb_section_pretty_'] = prettify_breadcrumb(get_module_name(func))
+                result['breadcrumb_section_'] = get_module_name(func)
+            except:
+                pass
+
+            if c.UBER_SHUT_DOWN and not cherrypy.request.path_info.startswith('/schedule'):
+                return render('closed.html')
+            elif isinstance(result, dict):
+                return render(_get_template_filename(func), result)
+            else:
+                return result
+        except CSRFException as e:
```

Comment:
I get what you're doing here, but putting a ton of code inside a single `try` statement is generally a red flag (and makes the code pretty hard to read). What was wrong with just putting this exception in the `csrf_protected` function?